browser-extension/mv3/src/service-worker/services/ruleExecutionHandler.ts

```diff
@@ -0,0 +1,57 @@
+import { Rule } from "common/types";
+import extensionIconManager from "./extensionIconManager";
+import { DataScope, TAB_SERVICE_DATA, tabService } from "./tabService";
+import { getRecords } from "common/storage";
+
+interface RulesExecutionLog {
+  ruleId: string;
+  requestDetails: chrome.webRequest.WebRequestDetails | chrome.webRequest.WebResponseDetails;
+}
+
+class RuleExecutionHandler {
+  constructor() {}
+
+  getExecutedRules = async (tabId: number) => {
+    const rulesExecutionLogs: RulesExecutionLog[] =
+      tabService.getPageData(tabId, TAB_SERVICE_DATA.RULES_EXECUTION_LOGS, []) || [];
+
+    const appliedRuleIds = rulesExecutionLogs.map((executionLog) => executionLog.ruleId).filter((id) => !!id);
```

Comment:
Why check for existence of ruleId?

Reply:
Just to be double sure. But yeah, this can be removed